lib/CAPI/Dialect/AIG.cpp

```diff
@@ -9,10 +9,129 @@
 #include "circt-c/Dialect/AIG.h"
 #include "circt/Dialect/AIG/AIGDialect.h"
 #include "circt/Dialect/AIG/AIGPasses.h"
+#include "circt/Dialect/AIG/Analysis/LongestPathAnalysis.h"
+#include "mlir-c/BuiltinAttributes.h"
+#include "mlir-c/IR.h"
+#include "mlir-c/Support.h"
 #include "mlir/CAPI/IR.h"
 #include "mlir/CAPI/Registration.h"
 #include "mlir/CAPI/Support.h"
+#include "mlir/Pass/AnalysisManager.h"
+#include "llvm/Support/JSON.h"
+
+using namespace circt;
+using namespace circt::aig;
 
 MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(AIG, aig, circt::aig::AIGDialect)
 
 void registerAIGPasses() { circt::aig::registerPasses(); }
+
+// Wrapper struct to hold both the analysis and the analysis manager
+struct LongestPathAnalysisWrapper {
+  std::unique_ptr<mlir::ModuleAnalysisManager> analysisManager;
+  std::unique_ptr<LongestPathAnalysis> analysis;
+};
+
+// Helper functions for wrapping/unwrapping
+static inline AIGLongestPathAnalysis wrap(LongestPathAnalysisWrapper *wrapper) {
+  return AIGLongestPathAnalysis{wrapper};
+}
+
+static inline LongestPathAnalysisWrapper *
+unwrap(AIGLongestPathAnalysis analysis) {
+  return static_cast<LongestPathAnalysisWrapper *>(analysis.ptr);
+}
+
+static inline AIGLongestPathCollection wrap(LongestPathCollection *collection) {
+  return AIGLongestPathCollection{collection};
+}
+
+static inline LongestPathCollection *
+unwrap(AIGLongestPathCollection collection) {
+  return static_cast<LongestPathCollection *>(collection.ptr);
+}
```

Comment:
Why isn't it possible to use the helper macro? https://github.com/llvm/llvm-project/blob/c7d3b818963aede5fc80a3246fb9af3f21f8eadd/mlir/include/mlir/CAPI/Wrap.h#L25